lib/base-compiler.js

```diff
@@ -55,6 +56,11 @@ Compile.prototype.newTempDir = function () {
 Compile.prototype.writeFile = Promise.denodeify(fs.writeFile);
 Compile.prototype.readFile = Promise.denodeify(fs.readFile);
 Compile.prototype.stat = Promise.denodeify(fs.stat);
+Compile.prototype.couldHaveOptOutput = function(options, version) {
+    //TODO(jared): a compiler meta data gatherer could improve this
+    return version.toLowerCase().indexOf("clang") > -1 && 
+          options.some((x) => x == "-fsave-optimization-record");
```

Comment:
There's no gcc/clang. But you can easily add a "supportsOptimization" flag not unlike the other flags (`options` or `compilerType` or similar). I wouldn't be against making `clang` a different `compilerType` too; which would then mean a whole set of features could be lumped in a new `clang.js` (which basically inherits from `base-compiler`). So many ways to achieve the same results...not at all sure what's best.